csv/csv_parse_stream.ts

```diff
@@ -41,8 +41,8 @@ class StreamLineReader implements LineReader {
     }
   }
 
-  isEOF(): Promise<boolean> {
-    return Promise.resolve(this.#done);
+  isEOF(): boolean {
+    return this.#done;
```

Comment:
Note: `StreamLineReader` (`LineReader` interface) is only used internally in `CsvParseStream`. So we can change this without breaking user facing APIs.